packages/cloud-cognitive/src/components/AddSelect/AddSelectFilter.js

```diff
@@ -54,77 +65,118 @@ export let AddSelectFilter = ({
-                  key={filterOpts.id}
-                  titleText={filterOpts.label}
-                  items={filterOpts.opts}
-                  light
-                  onChange={(value) => onchangeHandler(value, filterOpts.id)}
-                  selectedItem={getSelectedItem(filterOpts.id)}
-                  label={placeholder}
-                />
-              ))}
+        {showFilter && (
+          <Button
+            renderIcon={Filter16}
+            hasIconOnly
+            kind="ghost"
+            onClick={() => setOpen(!open)}
+            iconDescription={iconDescription}
+            className={`${blockClass}-toggle`}
+            size="md"
+          />
+        )}
+        {open && (
+          <div className={`${blockClass}`}>
+            <div className={`${blockClass}-content`}>
+              <p>Filters</p>
+              <div className={`${blockClass}-opts`}>
+                {filterOpts.map((filterOpts) => (
+                  <Dropdown
+                    id={filterOpts.id}
+                    key={filterOpts.id}
+                    titleText={filterOpts.label}
+                    items={filterOpts.opts}
+                    light
+                    onChange={(value) => onchangeHandler(value, filterOpts.id)}
+                    selectedItem={getSelectedItem(filterOpts.id)}
+                    label={placeholder}
+                  />
+                ))}
+              </div>
             </div>
+            <ButtonSet className={`${blockClass}-button-set`}>
+              <Button
+                kind="secondary"
+                onClick={resetFilters}
+                className={`${blockClass}-button`}
+              >
+                {secondaryButtonText}
+              </Button>
+              <Button
+                kind="primary"
+                onClick={applyFilters}
+                className={`${blockClass}-button`}
+              >
+                {primaryButtonText}
+              </Button>
+            </ButtonSet>
           </div>
-          <ButtonSet className={`${blockClass}-button-set`}>
-            <Button
-              kind="secondary"
-              onClick={resetFilters}
-              className={`${blockClass}-button`}
-            >
-              {secondaryButtonText}
-            </Button>
-            <Button
-              kind="primary"
-              onClick={applyFilters}
-              className={`${blockClass}-button`}
+        )}
+      </div>
+      {hasFiltersApplied && (
+        <div className={`${blockClass}-applied`}>
+          {Object.keys(appliedFilters).map((filterType) => (
+            <Tag
+              key={filterType}
+              type="gray"
+              size="sm"
+              onClose={() => removeTag(filterType)}
+              filter
             >
-              {primaryButtonText}
-            </Button>
-          </ButtonSet>
+              {`${filterType}: ${appliedFilters[filterType]}`}
+            </Tag>
+          ))}
+          <span
```

Comment:
I think would be best to use a ghost or tertiary button here. Also with the way you have `onKeyDown` currently, all the filters are cleared out when you press any key. I'm not sure that's the intended behavior.